giraffe/src/components/TableCell.tsx

```diff
@@ -0,0 +1,325 @@
+// Libraries
+import React, {FunctionComponent} from 'react'
+import classnames from 'classnames'
+
+// Constants
+import {ASCENDING, DEFAULT_TIME_FIELD} from '../constants/tableGraph'
+
+// Utils
+import {generateThresholdsListHexs} from '../utils/colorOperations'
+import {isString} from '../utils/isString'
+import {defaultTo} from '../utils/defaultTo'
+
+// Types
+import {TableViewProperties, SortOptions, RenamableField} from '../types'
+import {CellRendererProps} from './TableGraphTable'
+
+// Styles
+import styles from './TableGraphs.scss'
+
+interface Props extends CellRendererProps {
+  sortOptions: SortOptions
+  data: string
+  dataType: string
+  properties: TableViewProperties
+  hoveredRowIndex: number
+  hoveredColumnIndex: number
+  isTimeVisible: boolean
+  isVerticalTimeAxis: boolean
+  isFirstColumnFixed: boolean
+  onClickFieldName: (data: string) => void
+  onHover: (e: React.MouseEvent<HTMLElement>) => void
+  resolvedRenamableFields: RenamableField[]
+  timeFormatter: (time: string) => string
+}
+
+const URL_REGEXP = /((http|https)?:\/\/[^\s]+)/g
+
+// NOTE: rip this out if you spend time any here as per:
+// https://stackoverflow.com/questions/1500260/detect-urls-in-text-with-javascript/1500501#1500501
+const asLink = (str: string) => {
+  const isURL = `${str}`.includes('http://') || `${str}`.includes('https://')
+  if (isURL === false) {
+    return str
+  }
+
+  const regex = RegExp(URL_REGEXP.source, URL_REGEXP.flags)
+  const out = []
+  let index = 0
+  let link
+  let match
+
+  do {
+    match = regex.exec(str)
+
+    if (match) {
+      if (match.index - index > 0) {
+        out.push(str.slice(index, match.index))
+      }
+
+      link = str.slice(match.index, match.index + match[1].length)
+      out.push(
+        <a href={link} target="_blank">
+          {link}
+        </a>
+      )
+
+      index = match.index + match[1].length
```

Comment:
match[1] happens to be safe here. It exists because of the RegEx substring matcher for "http" or "https". If either of those substrings do not exist, then the early return prevents this code from being run.